### ISSUE TITLE
[explicit-function-return-type] false positives with typed function expressions in interfaces

### ISSUE THREAD

**Issue by busybeaver** [typescript-eslint/typescript-eslint#408]
The _explicit-function-return-type_ rule, with _allowTypedFunctionExpressions=true_ set, triggers if the typed function expression is a property in an interface definition (and not "just" a single/standalone function).

**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/explicit-function-return-type": ["warn", { "allowTypedFunctionExpressions": true }],
  }
}
```

```TS
interface Test {
  foo(): Promise<boolean>;
  bar(input: string): number;
}

// results in linter warnings
const implementation: Test = {
  foo: async () => true,
  bar: (input: string) => 42,
};

// works fine without the linter complaining...
const foo: Test['foo'] = async () => true;
```

**Expected Result**

The _explicit-function-return-type_ rule (with _allowTypedFunctionExpressions=true_) should not complain/be triggered by the code above.

**Actual Result**

The rule identifies the code as non-compliant.

![image](https://user-images.githubusercontent.com/858259/55634939-d545c400-57bf-11e9-84b4-2f0f6aeaf64d.png)

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `1.6.0` |
| `@typescript-eslint/parser`        | `1.6.0` |
| `TypeScript`                       | `3.4.1` |
| `ESLint`                           | `5.16.0` |
| `node`                             | `10.13.0` |
| `npm`                              | `6.4.1` |

**Comment by deser** [comment]
Also its very annoying for react components

**Comment by bradzacher** [comment]
@deser - this shouldn't affect react components.
Try using the `allowTypedFunctionExpressions` option (see the docs for the rule).

**Comment by deser** [comment]
You mean `allowExpressions` as stated here https://github.com/bradzacher/eslint-plugin-typescript/blob/master/docs/rules/explicit-function-return-type.md?

**Comment by bradzacher** [comment]
Should turn on both!
